scripts/build-game.sh fails on lsp-types

System: Ubuntu 26.04, aarch64

From `main` branch (commit `14f80096`), ran:
```
scripts/build-game.sh
```

Tried with both GHC 9.6.7 and GHC 9.8.4, but got the same error:

```
src/Language/LSP/Protocol/Message/Method.hs:64:10: error: [GHC-39999]
    • No instance for ‘Data.EqP.EqP SMethod’
        arising from the superclasses of an instance declaration
    • In the instance declaration for ‘GEq SMethod’
   |
64 | instance GEq SMethod where
   |          ^^^^^^^^^^^

src/Language/LSP/Protocol/Message/Method.hs:72:10: error: [GHC-39999]
    • No instance for ‘Data.OrdP.OrdP SMethod’
        arising from the superclasses of an instance declaration
    • In the instance declaration for ‘GCompare SMethod’
   |
72 | instance GCompare SMethod where
   |          ^^^^^^^^^^^^^^^^

[...]
Failed to build lsp-types-2.1.0.0 (which is required by exe:swarm from swarm-0.8.0.0). See the build log above for details.
```

Incidentally, `lsp` is a rather large dependency.  Should we really require it as a dependency of `swarm-lang`?

--- a/cabal.project
+++ b/cabal.project
@@ -2,4 +2,5 @@ packages: .
 
 constraints:
   # Required for build success with ghc-9.6 and ghc-9.8
+  # See https://github.com/swarm-game/swarm/issues/2713
   lsp-types >= 2.1.1

--- a/cabal.project
+++ b/cabal.project
@@ -1,0 +1,5 @@
+packages: .
+
+constraints:
+  # Required for build success with ghc-9.6 and ghc-9.8
+  lsp-types >= 2.1.1

